sharktank/tests/models/llama/benchmark_tests.py

```diff
@@ -0,0 +1,779 @@
+            tensor_parallelism_size,
+            irpa_path,
+            output_mlir_path,
+            output_json_path,
+        )
+        logging.getLogger().info(f"Launching export command:\n" f"cd {cwd} && {cmd}")
+        proc = subprocess.run(cmd, shell=True, capture_output=True, cwd=cwd)
+        return_code = proc.returncode
+        if return_code != 0:
+            raise Exception(f"{cmd} failed to export.")
+
+    def iree_compile(
+        self, mlir_path: str, output_file: str, args: List[str], cwd: str | Path
+    ):
+        """Compiles an input MLIR file to an output .vmfb file.
+        This assumes that the `iree-compile` command is available (usually via PATH).
+        Args:
+            mlir_path: Path to the input MLIR file.
+            output_file: Path for the output file. The directory must already exist.
+            args: List of arguments to pass to `iree-compile`.
+            cwd: current working directory
+        Raises Exception if compilation fails for some reason.
+        """
+        cmd = self.get_compile_cmd(mlir_path, output_file, args)
+        logging.getLogger().info(f"Launching compile command:\n" f"cd {cwd} && {cmd}")
+        proc = subprocess.run(cmd, shell=True, capture_output=True, cwd=cwd)
+        return_code = proc.returncode
+        if return_code != 0:
+            raise Exception(f"{cmd} failed to compile.")
+
+    def iree_benchmark_module(
+        self,
+        hip_device_id: str,
+        vmfb_name: str,
+        irpa_path: str,
+        args: List[str],
+        cwd: str | Path,
+    ):
+        """Runs a compiled program with the given args using `iree-benchmark-module`.
+        This assumes that the `iree-benchmark-module` command is available (usually via PATH).
+        Args:
+            vmfb_name: Name of the .vmfb file (relative to `cwd`).
+            args: List of arguments to pass to `iree-benchmark-module`.
+            cwd: Working directory to run the command within. (either string or Path works)
+            compile_cmd: Command used to compile the program, for inclusion in error messages.
+        Raises Exception if running fails for some reason.
+        """
+        benchmark_args = [
+            f"ROCR_VISIBLE_DEVICES={hip_device_id}",
+            "iree-benchmark-module",
+            f"--device=hip://{hip_device_id}",
+            "--hip_use_streams=true",
+            "--hip_allow_inline_execution=true",
+            "--device_allocator=caching",
+            f"--module={vmfb_name}",
+            f"--parameters=model={irpa_path}",
+        ]
+        benchmark_args += args
+        cmd = subprocess.list2cmdline(benchmark_args)
+        logging.getLogger().info(f"Launching run command:\n" f"cd {cwd} && {cmd}")
+        proc = subprocess.run(cmd, shell=True, stdout=sys.stdout, cwd=cwd)
+        return_code = proc.returncode
+        if return_code != 0:
+            raise Exception(f"{cmd} failed to run")
+
+    def cleanup_output_files(
```

Comment:
probably better to use tempfile like done here https://github.com/nod-ai/SHARK-Platform/blob/0c2e965c3ffe723db2fe2be9193c6d45fe558dbe/sharktank/tests/types/tensors_test.py#L71. Should better handle potential multiple async runs effectively corrupting the data in these effectively hardcoded paths

Reply:
We want to save the output files to track any regressions right? So, do we still need tempfile? I think `os.mkdir()` would make more sense to use in this case, especially when specifying the directory path of the artifiacts to upload.

Reply:
you can upload the output artifacts before tempfile gets rid of them. I don't think we want to keep the files on the CI system permanently. os.mkdir could still have issues with multiple simultaneous instances running interfering with each other unless you are very careful with it

Reply:
How can I upload the output artifacts before tempfile gets rid of them?

Reply:
https://github.com/nod-ai/SHARK-TestSuite/blob/95b35cc076618ef79b7f84dc4117f4528451e5e6/e2eshark/_run_helper.py#L61 for example uploading files. You should just need to do that before closing the tempfile or directory that is created

Reply:
Do we prefer sharkpublic? We will be uploading a bunch of artifacts there every night, and it might be hard to correlate which run correlates to which files. With github actions, we can just click on a run and see it's respective output artifacts without cluttering the storage

Reply:
Does github actions work well for downloading things like vmfbs? If so, then that would work well.

Reply:
How much data are we talking about here? Cloud storage isn't cheap. Using GitHub artifacts with GitHub workflows is nice since it doesn't require separate privileges to maintain, but then you're limited by GitHub's retention policies.

If you want to keep historical data in an organized way, please design a database early. I've seen too many benchmarking systems without that structure and it makes analysis tricky.

For this PR itself, I'd try to keep the scope small and checkpoint the work, rather than get hung up on all these discussions. Fine to land and iterate a bit so things don't pile up.

Reply:
Yeah let's start with github artifacts without a temp dir (it won't persist between runs anyways) and iterate from there